ompi/mca/hook/comm_method/hook_comm_method_fns.c

```diff
@@ -364,6 +370,26 @@ ompi_report_comm_methods(int called_from_location) // 1 = from init, 2 = from fi
         return;
     }
 
+    char prefix_string[64];
+    char *unprefixed_affstring;
+    prefix_string[0] = 0;
+    if (mylocalrank == 0) {
+        myleaderrank = ompi_comm_rank(leader_comm);
+        sprintf(prefix_string, "H%d: ", myleaderrank);
+    }
+    int indentation = strlen(prefix_string);
+    /* reject really small aff_columns settings*/
+    if (mca_hook_comm_method_aff_columns < indentation + 16) {
+        mca_hook_comm_method_aff_columns = indentation + 16;
+    }
+    host_local_get_affstring(&unprefixed_affstring, indentation,
+        mylocalrank, nlocalranks, local_comm);
+    if (mylocalrank == 0) {
+        affstring = malloc(strlen(prefix_string) + strlen(unprefixed_affstring) + 16);
+        sprintf(affstring, "%s%s", prefix_string, unprefixed_affstring);
```

Comment:
can you switch this to snprintf(), or perhaps even asprintf()? This would guard against overflows a bit better, unlikely as it may be.